go/ql/lib/semmle/go/dataflow/ExternalFlow.qll

```diff
@@ -213,16 +213,30 @@ predicate interpretModelForTest(QlBuiltins::ExtensionId madId, string model) {
   )
 }
 
+bindingset[p]
+private string cleanPackage(string p) {
+  exists(string noPrefix |
+    p = fixedVersionPrefix() + noPrefix
+    or
+    not p = fixedVersionPrefix() + any(string s) and
+    noPrefix = p
+  |
+    result = noPrefix.regexpReplaceAll(majorVersionSuffixRegex(), "")
+  )
+}
+
 private predicate relevantPackage(string package) {
-  sourceModel(package, _, _, _, _, _, _, _, _, _) or
-  sinkModel(package, _, _, _, _, _, _, _, _, _) or
-  summaryModel(package, _, _, _, _, _, _, _, _, _, _)
+  exists(string p | package = cleanPackage(p) |
+    sourceModel(p, _, _, _, _, _, _, _, _, _) or
+    sinkModel(p, _, _, _, _, _, _, _, _, _) or
+    summaryModel(p, _, _, _, _, _, _, _, _, _, _)
+  )
 }
 
 private predicate packageLink(string shortpkg, string longpkg) {
   relevantPackage(shortpkg) and
   relevantPackage(longpkg) and
-  longpkg.prefix(longpkg.indexOf(".")) = shortpkg
+  longpkg.prefix(longpkg.indexOf("/")) = shortpkg
```

Comment:
? How did this ever work?

Reply:
It didn't. This stuff was all written for java and C# and ported to go without fully unerstanding what it did. It's only used for making the automated library coverage reports, so we were getting `os` and `os/exec` as separate things when we could have had them counted together.